### PR TITLE
Fixed skipFirstLevel project parameter for factories

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Fix Factory validator to allow skipFirstLevel parameter (for skipping first directory while importing a ZIP project)

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10464

### Docs PR
https://github.com/eclipse/che-docs/pull/461

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]
ok if tests will pass

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/impl/SourceStorageParametersValidator.java [hunk 1]
```diff
@@ -41,6 +41,16 @@ public void validate(SourceStorage source, FactoryParameter.Version version)
                     entry.getValue()));
           }
           break;
+        case "skipFirstLevel":
+          final String skipFirstLevel = entry.getValue();
+          if (!"true".equals(skipFirstLevel) && !"false".equals(skipFirstLevel)) {
+            throw new ConflictException(
+                format(
+                    PARAMETRIZED_ILLEGAL_PARAMETER_VALUE_MESSAGE,
+                    "source.project.parameters.skipFirstLevel",
```
Comment (sleshchenko) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/impl/SourceStorageParametersValidator.java]: As far as I see there is not `project` field in `source` object while `project` object has `source` field. Should not it be `"project.source.parameters.skipFirstLevel"`?

**Comment by skabashnyuk** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/634/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/634//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10837)
docker image: eclipseche/che-server:10837
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by mkuznyetsov** [comment]
Docs PR: https://github.com/eclipse/che-docs/pull/461

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/639/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/639//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10837)
docker image: eclipseche/che-server:10837
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by skabashnyuk — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/impl/SourceStorageParametersValidator.java (hunk 1 above)
Comment (mkuznyetsov) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/impl/SourceStorageParametersValidator.java]: ty, fixed

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/639/) doesn't show any regression against this Pull Request

**Pull request merged**